20250226-actors-pubsub.md

```diff
@@ -0,0 +1,158 @@
+# Actor PubSub Implementation Proposal
+
+- Author(s): @joshvanl
+
+## Overview
+
+This proposal introduces PubSub capabilities for Dapr Actors, enabling actors to subscribe to messages for their actor type.
+Additionally, actors of a that type have the ability to publish messages to that type, to be delivered to  particular actor IDs.
+A single designated PubSub component will be marked as the actor PubSub for that namespace, similar to how an actor state store is defined.
+There can be only one actor PubSub Component per namespace.
+Actors will be spawned, if they are not already, when they receive an actor PubSub message.
+
+## Background
+
+Today, Dapr provides PubSub functionality for applications but lacks native PubSub integration for actors.
+This limitation means that actors cannot communicate efficiently via publish-subscribe messaging patterns within their type.
+By enabling Actor PubSub, actors can publish messages to specific actor IDs, improving scalability and communication reliability.
+
+### Considerations
+
+There are a number of scenarios or use cases which could benefit from an "Actor PubSub" feature.
+These include implicitly or explicitly subscribing to topics as an actor, whether publishing messages will spawn actors, if the message is broadcast to a single or all active actors, and if multiple topics can exist within a single actor type domain.
+This proposal describes a single topic per actor type that is implicitly subscribed to by all actors of that type.
+Other scenarios can be considered in future iterations.
+
+## Implementation Details
+
+### Component
+
+A single PubSub component will be designated as the actor PubSub, similar to how an actor state store is configured.
+It is intentional there can only be a single PubSub to a namespace to ensure that infrastructure is completely abstracted from actor developer consumers.
+It is also the case having multiple PubSubs would complicate subscribing, forcing actors to watch for PubSub resources and resubscribe as necessary.
+It seems logical to keep parity of single Component for both state and PubSub messaging.
+
+The metadata configuration will follow the existing state store pattern:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+ name: joshvanl
+spec:
+ type: pubsub.joshvanl
+ version: v1
+ metadata:
+ - name: actorpubsub # Added metadata field for all PubSubs.
```

Comment:
Generally, Im ok with this proposal as a starting point to be expanded upon. Just a few clarification questions:

Could you provide an example or 2 for this feature from an end user perspective and corresponding diagram depicting it in action? 

Does the `actorPubSub` inherit the pubsub resiliency policies if available or do we need a resiliency policy for this? I see we have a `pubsub` and `actor` [target](https://docs.dapr.io/operations/resiliency/targets/). 

Similarly, what happens with in-flight messages if an actor instance is deactivated while processing? Should actor deactivation wait for in-flight messages to complete?